ddl/column.go

```diff
@@ -841,11 +880,12 @@ func (w *worker) doModifyColumnTypeWithData(
 			}
 			if kv.ErrKeyExists.Equal(err) || errCancelledDDLJob.Equal(err) || errCantDecodeIndex.Equal(err) {
 				logutil.BgLogger().Warn("[ddl] run modify column job failed, convert job to rollback", zap.String("job", job.String()), zap.Error(err))
-				// TODO: Do rollback.
+				// When encounter these error above, we change the job to rolling back job directly.
+				job.State = model.JobStateRollingback
+				return ver, errors.Trace(err)
 			}
 			if types.ErrOverflow.Equal(err) {
-				// TODO: Do rollback.
-				job.State = model.JobStateCancelled
+				job.State = model.JobStateRollingback
```

Comment:
Merge line 887 and 881 into `1` if statement?

Reply:
addresssed

Reply:
addressed